process/elasticproc/block/blockProcessor_test.go

```diff
@@ -274,3 +288,132 @@ func TestBlockProcessor_PrepareBlockForDBEpochStartMeta(t *testing.T) {
 		DeveloperFees:     "0",
 	}, dbBlock)
 }
+
+func TestBlockProcessor_PrepareBlockForDBMiniBlocksDetails(t *testing.T) {
+	t.Parallel()
+
+	gogoMarshaller := &marshal.GogoProtoMarshalizer{}
+	bp, _ := NewBlockProcessor(&mock.HasherMock{}, &mock.MarshalizerMock{})
+
+	mbhr := &dataBlock.MiniBlockHeaderReserved{
+		IndexOfFirstTxProcessed: 0,
+		IndexOfLastTxProcessed:  0,
+	}
+	mbhrBytes, _ := gogoMarshaller.Marshal(mbhr)
+
+	txHash, notExecutedTxHash, invalidTxHash, rewardsTxHash, scrHash := "tx", "notExecuted", "invalid", "reward", "scr"
+	dbBlock, err := bp.PrepareBlockForDB([]byte("hash"), &dataBlock.Header{
+		TxCount: 5,
+		MiniBlockHeaders: []dataBlock.MiniBlockHeader{
+			{
+				TxCount:  1,
+				Type:     dataBlock.TxBlock,
+				Reserved: mbhrBytes,
+			},
+			{
+				TxCount: 1,
+				Type:    dataBlock.RewardsBlock,
+			},
+			{
+				TxCount: 1,
+				Type:    dataBlock.InvalidBlock,
+			},
+			{
+				TxCount: 1,
+				Type:    dataBlock.SmartContractResultBlock,
+			},
+		},
+	}, nil, &dataBlock.Body{
+		MiniBlocks: []*dataBlock.MiniBlock{
+			{
+				Type:     dataBlock.TxBlock,
+				TxHashes: [][]byte{[]byte(txHash), []byte(notExecutedTxHash)},
+			},
+			{
+				Type:     dataBlock.RewardsBlock,
+				TxHashes: [][]byte{[]byte(rewardsTxHash)},
+			},
+			{
+				Type:     dataBlock.InvalidBlock,
+				TxHashes: [][]byte{[]byte(invalidTxHash)},
+			},
+			{
+				Type:     dataBlock.SmartContractResultBlock,
+				TxHashes: [][]byte{[]byte(scrHash)},
+			},
+		},
+	}, nil, outport.HeaderGasConsumption{}, 0, &outport.Pool{
+		Txs: map[string]coreData.TransactionHandlerWithGasUsedAndFee{
+			txHash: &outport.TransactionHandlerWithGasAndFee{
+				ExecutionOrder: 2,
+			},
+			notExecutedTxHash: &outport.TransactionHandlerWithGasAndFee{
+				ExecutionOrder: 0,
+			},
+		},
+		Rewards: map[string]coreData.TransactionHandlerWithGasUsedAndFee{
+			rewardsTxHash: &outport.TransactionHandlerWithGasAndFee{
+				ExecutionOrder: 3,
+			},
+		},
+		Invalid: map[string]coreData.TransactionHandlerWithGasUsedAndFee{
+			invalidTxHash: &outport.TransactionHandlerWithGasAndFee{
+				ExecutionOrder: 1,
+			}},
+		Scrs: map[string]coreData.TransactionHandlerWithGasUsedAndFee{
+			scrHash: &outport.TransactionHandlerWithGasAndFee{
+				ExecutionOrder: 0,
+			},
+		},
+	})
+	require.Nil(t, err)
+
+	require.Equal(t, &data.Block{
```

Comment:
it is hard to follow that the expected value is correct, maybe think of a way to refactor in a future PR (e.g create a function to return expected block based on an input)

Reply:
I will add a task for this refactor of the tests.